cardano-cli/src/Cardano/CLI/Shelley/Util.hs

```diff
@@ -0,0 +1,32 @@
+-- | Shelley CLI shared Utility functions
+module Cardano.CLI.Shelley.Util
+  (
+    displayErrorText
+  , rescue
+
+  , getNetworkId
+  , getNetworkId'
+  ) where
+
+import           Control.Monad.Trans.Except.Extra (firstExceptT, newExceptT)
+import           Data.Text as Text (pack)
+
+import           Cardano.Api
+import           Cardano.Prelude
+
+import           Cardano.CLI.Shelley.Commands (NetworkIdArg(..) )
+
+displayErrorText :: Error e => e -> Text
+displayErrorText = Text.pack . displayError
+
+rescue :: Functor m =>  m (Either x a) -> (x -> y) -> ExceptT y m a
+rescue action except = firstExceptT except $ newExceptT action
+
+getNetworkId' :: NetworkIdArg -> IO (Either EnvNetworkIdError NetworkId)
+getNetworkId' arg = case arg of
+  NetworkIdFromCLI x -> return $ Right x
+  NetworkIdFromEnv -> readEnvNetworkId
+
+getNetworkId :: NetworkIdArg -> (EnvNetworkIdError -> e) -> ExceptT e IO NetworkId
```

Comment:
We should export `getNetworkId'` as `getNetworkId`. Leave it as a vanilla function with no `ExceptT` usage. Then we should rename `getNetworkId` to `firstGetNetworkId` or something like that to make it more obvious that we are transforming the error.

Reply:
Primed names can be problematic in general, but here is no danger of ever using `getNetworkId'` instead of `getNetworkId` because they have different types.
`getNetworkId` is the primary function with 47 call sites.
`getNetworkId'` an auxiliary function with 7 call sites.
The simple name should be used for the primary function with more call sites
and the more complex name for the auxiliary function with fewer call sites.

Reply:
`getNetworkId` is specialized for our use case in the cli. This should be renamed to indicate we are mapping over the error hence the name `firstGetNetworkId`. These functions are likely to end up in `cardano-api`, which is why I am suggesting we rename `getNetworkId' :: NetworkIdArg -> IO (Either EnvNetworkIdError NetworkId)` to `getNetworkId`. We should not impose our specific cli design decisions (`ExceptT`) on users.